src/electionguard/big_integer.py

```diff
@@ -0,0 +1,110 @@
+from typing import Any, Tuple, Union
+from base64 import b16decode
+
+# pylint: disable=no-name-in-module
+from gmpy2 import mpz
+
+
+def _hex_to_int(input: str) -> int:
+    """Given a hex string representing bytes, returns an int."""
+    return int(input, 16)
+
+
+def _int_to_hex(input: int) -> str:
+    """Given an int, returns a hex string representing bytes."""
+    hex = format(input, "02X")
+    if len(hex) % 2:
+        hex = "0" + hex
+    return hex
+
+
+_zero = mpz(0)
+
+
+def _convert_to_element(data: Union[int, str]) -> Tuple[str, int]:
+    """Convert element to consistent types"""
+    if isinstance(data, str):
+        hex = data
+        integer = _hex_to_int(data)
+    else:
+        hex = _int_to_hex(data)
+        integer = data
+    return (hex, integer)
+
+
+class BigInteger(str):
+    """A specialized representation of a big integer in python"""
+
+    _value: mpz = _zero
+
+    def __new__(cls, data: Union[int, str]):  # type: ignore
+        (hex, integer) = _convert_to_element(data)
+        big_int = super(BigInteger, cls).__new__(cls, hex)
+        big_int._value = mpz(integer)
+        return big_int
+
+    @property
+    def value(self) -> mpz:
+        """Get internal value for math calculations"""
+        return self._value
+
+    def __int__(self) -> int:
+        """Overload int conversion."""
+        return int(self.value)
+
+    def __eq__(self, other: Any) -> bool:
+        """Overload == (equal to) operator."""
+        return (
+            isinstance(other, BigInteger) and int(self.value) == int(other.value)
```

Comment:
Do you want to add a check for a str type as well or will the BigInteger take care of that?

Reply:
We could add one, but I don't know if we really need one. It would essentially be a hex checker.